File: backend/app/schemas/channel.py
from typing import Literal
from pydantic import BaseModel, Field


ChannelInputType = Literal["youtube_url", "channel_name"]


class ChannelResolveRequest(BaseModel):
    query: str = Field(..., min_length=1)
    max_results: int = Field(default=8, ge=1, le=50)


class ChannelCandidate(BaseModel):
    channel_id: str | None = None
    name: str
    handle: str | None = None
    url: str
    description: str | None = None
    thumbnail_url: str | None = None
    subscriber_count_text: str | None = None
    subscriber_count: int | None = None
    video_count_text: str | None = None
    video_count: int | None = None
    verified: bool = False
    match_score: float = 0
    source: str


class ChannelResolveResponse(BaseModel):
    query: str
    input_type: ChannelInputType
    candidates: list[ChannelCandidate]
    selected_channel: ChannelCandidate | None = None
    message: str

File: frontend/src/types/channel.ts
export type ChannelMatchSource = "url" | "name" | "youtube_url" | "channel_name";

export type BackendChannelCandidate = {
  channel_id: string | null;
  name: string;
  handle: string | null;
  url: string;
  description: string | null;
  thumbnail_url: string | null;
  subscriber_count_text: string | null;
  subscriber_count: number | null;
  video_count_text: string | null;
  video_count: number | null;
  verified: boolean;
  match_score: number;
  source: string;
};

export type BackendChannelResolveResponse = {
  query: string;
  input_type: "youtube_url" | "channel_name";
  candidates: BackendChannelCandidate[];
  selected_channel: BackendChannelCandidate | null;
  message: string;
};

export type ChannelCandidate = {
  id: string;
  channelId: string | null;
  name: string;
  handle: string;
  url: string;
  avatarUrl: string;
  thumbnailUrl: string | null;
  description: string;
  subscriberCount: string;
  subscriberCountValue: number | null;
  videoCount: number | null;
  videoCountText: string;
  verified: boolean;
  matchScore: number;
  source: string;
  raw: BackendChannelCandidate;
};
